backends/urls.py

```diff
@@ -19,7 +21,9 @@
 
     # These three go to views in social_core
     # authentication / association
-    url(r'^login/(?P<backend>[^/]+){0}$'.format(extra), views.auth,
+
+    url(r'^login/(?P<backend>[^/]+){0}$'.format(extra),
+        RedirectView.as_view(url='/oauth_maintenance') if settings.OAUTH_MAINTENANCE_MODE else views.auth,
```

Comment:
I think this would be better as `pattern_name='oauth_maintenance'` instead of `url=?`